lib/Lib.thy

```diff
@@ -2729,4 +2729,12 @@ text \<open>Prevent clarsimp and others from creating Some from not None by fold
 definition
   "not_None x = (x \<noteq> None)"
 
+primrec findIndex' :: "nat \<Rightarrow> ('a \<Rightarrow> bool) \<Rightarrow> 'a list \<Rightarrow> nat option" where
+  "findIndex' _ _ [] = None"
+| "findIndex' n P (x#xs) = (if P x then Some n else findIndex' (Suc n) P xs)"
+
+definition
+  findIndex :: "('a \<Rightarrow> bool) \<Rightarrow> 'a list \<Rightarrow> nat option" where
+  "findIndex = findIndex' 0"
+
```

Comment:
I am not sure if we want to keep camel case for these, now that they are in Lib... @lsf37 ? Any preference?